### PR TITLE
Deprecating font caching mechanism + support for .pkl font files definitions

### PR DESCRIPTION
**Pull request by Lucas-C**
_cf._ #345

**Checklist**:

- [x] The GitHub pipeline is OK (green)

- [x] Unit tests are covering the code added / modified by this PR

- [x] This PR is ready to be merged

- [NA] In case of a new feature, docstrings have been added, with also some documentation in the `docs/` folder

- [x] A mention of the change is present in `CHANGELOG.md`

### CODE REVIEW

File: test/fonts/test_add_font.py [hunk 1]
```diff
@@ -10,20 +10,13 @@
...
-
-def test_add_font_non_existing_file_unicode():
-    pdf = FPDF()
-    with pytest.raises(FileNotFoundError) as error:
-        pdf.add_font("non-existing", uni=True)
-    expected_msg = "TTF Font file not found: non-existing.pkl"
-    assert str(error.value) == expected_msg
+    for uni in (True, False):
+        with pytest.raises(FileNotFoundError) as error:
+            pdf.add_font("non-existing", uni=uni)
```
Comment (Spenhouet) [inline test/fonts/test_add_font.py:17]: Does testing for the uni parameter still makes sense when it is deprecated without function?

**Review by Spenhouet — CHANGES_REQUESTED** [review]
The change looks good to me. Only one small remark on a test.

File: test/fonts/test_add_font.py [hunk 2]
```diff
@@ -54,7 +61,7 @@ def test_add_font_unicode_with_path_fname_ok(tmp_path):
     for font_cache_dir in (True, tmp_path, None):
         pdf = FPDF(font_cache_dir=font_cache_dir)
         font_file_path = HERE / "Roboto-Regular.ttf"
-        pdf.add_font("Roboto-Regular", fname=str(font_file_path))
+        pdf.add_font("Roboto-Regular", fname=str(font_file_path), uni=True)
```
Comment (Spenhouet) [inline test/fonts/test_add_font.py:64]: Was it intentional to reintroduce the uni parameter here?

Comment (Lucas-C) [reply test/fonts/test_add_font.py:64]: No, this was a mistake, thanks!

**Comment by Lucas-C** [comment]
Thansk for the review @Spenhouet!

**Pull request merged**